### PR TITLE
feat(builder): scaffold playwright

### PR DESCRIPTION
**Pull request by ntindle**
### Background

We need to do more robust testing of our services and platforms. Its clear due to some major breakages that have been occurring

### Changes 🏗️

Scaffolds out playwright and adds it to the CI, with a VERY simple test to evaluate that it works

Only for the new autogpt platform, currently in rnd/

- Create from scratch and execute an agent with at least 3 blocks
- Import an agent from file upload, and confirm it executes correctly
- Upload agent to marketplace
- Import an agent from marketplace and confirm it executes correctly
- Edit an agent from monitor, and confirm it executes correctly

### CODE REVIEW

**Review by Swiftyos — APPROVED** [review]
Tested it locally and it worked - though Microsoft Edge did not. Do I need to install that manually myself?

File: autogpt_platform/frontend/sentry.client.config.ts [hunk 1]
```diff
@@ -31,14 +31,6 @@ Sentry.init({
...
   ],
 
-  beforeSend(event, hint) {
-    // Check if it is an exception, and if so, show the report dialog
-    if (event.exception && event.event_id) {
-      Sentry.showReportDialog({ eventId: event.event_id });
-    }
-    return event;
-  },
-
```
Comment (Pwuts) [inline autogpt_platform/frontend/sentry.client.config.ts]: unrelated & undeclared

File: autogpt_platform/frontend/package.json [hunk 2]
```diff
@@ -8,7 +8,10 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
-    "format": "prettier --write ."
+    "format": "prettier --write .",
+    "test": "playwright test",
+    "test-ui": "playwright test --ui",
```
Comment (Pwuts) [inline autogpt_platform/frontend/package.json:14]: what's the difference between these two?

any chance you could add a bit of info about this playwright setup in the (advanced) README?

File: .github/workflows/platform-frontend-ci.yml [hunk 3]
```diff
@@ -2,40 +2,82 @@ name: AutoGPT Platform - Frontend CI
...
-        npm run lint
+      - uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "21"
+
+      - name: Install dependencies
+        run: |
+          npm install
```
Comment (Pwuts) [inline .github/workflows/platform-frontend-ci.yml:32]: nit:
```suggestion
        run: npm install
```

same below

File: autogpt_platform/frontend/sentry.client.config.ts (hunk 1 above)
Comment (ntindle) [reply autogpt_platform/frontend/sentry.client.config.ts]: was a sidestream brought in locally, worry not for it shall disappear

File: autogpt_platform/frontend/package.json (hunk 2 above)
Comment (ntindle) [reply autogpt_platform/frontend/package.json:14]: when its ready yeah, gotta get other stuff workin first

File: .github/workflows/platform-frontend-ci.yml (hunk 3 above)
Comment (ntindle) [reply .github/workflows/platform-frontend-ci.yml:32]: keeping current pattern, sorry <3. We need to update this across the workflows

**Pull request merged**